src/IB/CIBMethod.cpp

```diff
@@ -1716,6 +1715,7 @@ CIBMethod::flagRegrid() const
 void
 CIBMethod::getFromInput(Pointer<Database> input_db)
 {
+    d_use_steady_stokes = input_db->getBoolWithDefault("use_steady_stokes", d_use_steady_stokes);
```

Comment:
I have largely moved away from using `WithDefault`.

Reply:
> One restarted test example is `IB/explicit_ex0.restart=300.mpirun=4.input`.
> 
> The string `restart=300` tells the test runner to do to things:
> 
> 1. run the test normally
> 2. after the first run, run the test again as (in this case) `mpirun -np 4 ./explicit_ex0 ./explicit_ex0.restart=300.mpirun=4.input ./restart/ 300`. When done correctly this overwrites the previous output file with a new one from the restarted run.
> 
> Hence, the primary differences are that the input file should specify the restart directory as `restart`, it must save a restart file at the (in this case) 300th time step, and there should be more than (in this case) 300 time steps so that the restarted run has something to do.
> 
> I usually write restart tests by deleting the first half of the output file and then making sure that I get the same results for the second half of the restart file with the restarted run.
> 
> Does this make sense? It should only take a few minutes to add a restart variant of a test once we have a non-restarted test written. Let me know if you run into trouble.

Yes, but restart outputs are really tricky. I have seen that if you output forces and torques on they can differ in 4th or 5th decimal place, especially when they arise from pressure forces. In multiphase flows, we tested a restarted run: In the first cycle everything matched with non restarted run, but in the second cycle the solver residual were slightly different from second iteration onwards, which caused slight change in the output. The change in the decimal changed from 4th to 8th, when we increased the grid resolution.

So a a word of caution about restart runs: Don't go Nazi on slight differences in the non-restarted and restarted runs. It really depends what you output.

Reply:
I agree. In this case I'm more worried about O(1) bugs, like whether or not CIBMethod is set up as steady-state stokes or not. Changing something within the solver tolerance is totally fine with me.

Reply:
ok, yes, I like you always test restart capability. That is very important to me.

Reply:
I have to because I write a lot of restart-related bugs in IBFEMethod :)